### DISCUSSION TITLE
The Serverless version (3.8.0) does not satisfy the "frameworkVersion" (2) in serverless.yml

### DISCUSSION THREAD

**Discussion by NitinMagdumCimpress** [serverless/serverless#10886] [Q&A]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

$ serverless deploy --stage dev
Environment: linux, node 16.14.2, framework 3.8.0, plugin 6.1.6, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
The Serverless version (3.8.0) does not satisfy the "frameworkVersion" (2) in serverless.yml
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1

### Service configuration (serverless.yml) content

```yaml
$ serverless deploy --stage dev
Environment: linux, node 16.14.2, framework 3.8.0, plugin 6.1.6, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
The Serverless version (3.8.0) does not satisfy the "frameworkVersion" (2) in serverless.yml
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1
```

### Command name and used flags

 serverless deploy

### Command output

```shell
$ serverless deploy --stage dev
Environment: linux, node 16.14.2, framework 3.8.0, plugin 6.1.6, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
The Serverless version (3.8.0) does not satisfy the "frameworkVersion" (2) in serverless.yml
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1
```

### Environment information

```shell
$ serverless deploy --stage dev
Environment: linux, node 16.14.2, framework 3.8.0, plugin 6.1.6, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
The Serverless version (3.8.0) does not satisfy the "frameworkVersion" (2) in serverless.yml
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1
```

**Comment by pgrzesik** [comment]
Hello @NitinMagdumCimpress - I'm moving it to discussions - the error message answers the problem - you have specified `frameworkVersion: 2` but you're using `v3` major version - update the `frameworkVersion` setting or downgrade to previous major of the Framework

**Reply by NitinMagdumCimpress** [reply]
After changing to frameworkVersion: "3" in serverless.yml getting below error

$ serverless deploy --stage dev
Deploying logoform-serverless-lambda1 to stage dev (eu-west-1)
× Stack logoform-serverless-lambda1-dev failed to deploy (0s)
Environment: linux, node 16.14.2, framework 3.8.0, plugin 6.1.6, SDK 4.3.2
Credentials: Local, environment variables
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
TypeError: this.serverless.variables.populateObject is not a function
    at /builds/vistaprint-org/design-technology/logo-diy/logoform-serverless-lambda/node_modules/serverless-step-functions/lib/yamlParser.js:26:[62](https://gitlab.com/vistaprint-org/design-technology/logo-diy/logoform-serverless-lambda/-/jobs/2234036055#L62)
4 deprecations found: run 'serverless doctor' for more details
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1

**Reply by pgrzesik** [reply]
It looks like you might be using outdated versions of the plugins that not support `v3` Framework - please ensure you've upgraded them too.

**Reply by NitinMagdumCimpress** [reply]
"serverless-prune-plugin": "^1.5.0",
    "serverless-pseudo-parameters": "^2.5.0",
    "serverless-step-functions": "^2.30.0"
    
    These extra packages we are using
    where can i find a document for supporting plugins for version 3.8.0

**Reply by pgrzesik** [reply]
You should visit repositories of each of the plugin and check if there were any releases aimed at supporting v3 - in general, the best way would be to upgrade them to the latest version.
